### PR TITLE
feat(es-dev-server): allow using the server as a library

### PR DESCRIPTION
**Pull request by LarsDenBakker**
Splits up `server.js` into separate parts:
- createConfig
- createMiddlewares
- createServer
- startServer

This will make it much easier to compose the dev server, and use it as a library.

Also adds documentation on how to use the dev server as a library in the advanced section of the docs.

Fixes https://github.com/open-wc/open-wc/issues/608

### CODE REVIEW

**Comment by LarsDenBakker** [comment]
Right now programmatically using the server uses a different config object than the command line args config. I want to align these first before merging.

**Comment by LarsDenBakker** [comment]
This is ready for review, please check if the added docs make any sense!

**Comment by LarsDenBakker** [comment]
@bennypowers these titles reflect the functions they're calling, I thought it'd be nice for findability

**Comment by bennypowers** [comment]
> @bennypowers these titles reflect the functions they're calling, I thought it'd be nice for recognition.

I was specifically referring to 

> @bennypowers these titles reflect the functions they're calling, I thought it'd be nice for recognition.

I meant specifically `Command line flags and configuration` => `Command Line Flags and Configuration`. I agree about e.g. `# \`createServer\``

**Comment by LarsDenBakker** [comment]
Ah yes, good catch. I've updated the readme!

File: packages/es-dev-server/README.md [hunk 1]
```diff
@@ -374,6 +380,96 @@ Compatibility mode enables bundle-free development with features such as es modu
...
+`startServer` creates and starts the server, listening on the configured port. It opens the browser if configured and logs a startup message.
+
+Returns the koa app and a node http or http2 server.
+
+```javascript
+import Koa from 'koa';
+import { createConfig, startServer } from 'es-dev-server';
+
+const config = createConfig({ ... });
+const { app, server } = startServer(config, fileWatcher);
```
Comment (kevinmpowell) [inline packages/es-dev-server/README.md]: @LarsDenBakker is there a callback that can be run after `startServer` is successful? My use case is for Visual Regression Testing. I need to:

1. Spin up the test server and wait for it to be ready
2. Seed my testing framework with the server's url/port and run the tests against the test server
3. Shut down the test server after my testing framework has finished running.

Comment (LarsDenBakker) [reply packages/es-dev-server/README.md]: I'm doubting whether it's not better to use `createServer` in that case. That way, you're in full control when it starts. What do you think?

Comment (kevinmpowell) [reply packages/es-dev-server/README.md]: It's not about being in control of when it starts, it's about knowing when the server is ready to receive requests. See Browsersync's `init` command. https://www.browsersync.io/docs/api#api-init

**Pull request merged**